src/Npgsql/MultiplexingDataSource.cs

```diff
@@ -349,6 +349,15 @@ static void CompleteWrite(NpgsqlConnector connector, ref MultiplexingStats stats
         // ReSharper disable once FunctionNeverReturns
     }
 
+    protected override void Dispose(bool disposing)
+    {
+        // Force doing the startup checks again when disposed
+        // in order to reset the Connection's DataSource
+        if (disposing)
+            StartupCheckPerformed = false;
```

Comment:
Can you provide a bit more context on why this is needed? AFAICT this flag is only consulted when opening the first connection on a multiplexing data source, I'm not sure why we'd need to reset it here.

Reply:
That's the simplest solution to make things work if you call ReloadTypes *after* opening the first multiplexing connection.
Without it, you fail in the next command's execution after calling it and from there it it very hard to recover the DataSource.